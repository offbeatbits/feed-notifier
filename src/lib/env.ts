/* istanbul ignore next */
import { config } from 'dotenv';
/* istanbul ignore next */
if (process.env.NODE_ENV === 'development') config();
