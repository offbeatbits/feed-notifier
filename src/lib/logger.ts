import {
  app,
} from '../config';

const {
  enableLogging,
  enableVerboseLogging,
} = app;

export const logger = {
  info: (content: string) => {
    if (enableLogging) {
      console.log(`[info] ${content}`); // eslint-disable-line
    }
  },
  verbose: (content: string) => {
    if (enableVerboseLogging) {
      console.log(`[verbose] ${content}`); // eslint-disable-line
    }
  },
};
