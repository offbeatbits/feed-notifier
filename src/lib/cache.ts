import fs from 'fs';
import path from 'path';
import { app } from '../config';
import { logger } from './logger';

const { cacheDirectory } = app;

const createDirectoryMaybe = (directory: string) => {
  if (!fs.existsSync(directory)) {
    fs.mkdirSync(directory);
  }
};

interface SaveToFileArgs {
  directory: string;
  file: string;
  content: string;
}

const saveToFile = ({
  directory,
  file,
  content,
}: SaveToFileArgs) => {
  createDirectoryMaybe(directory);
  const filePath = path.join(directory, file);
  fs.writeFileSync(filePath, content);
};

const getFileContents = (filePath: string) =>
  Buffer.from(fs.readFileSync(filePath)).toString() || '';

export class Cache {
  static get(name: string) {
    logger.verbose(`Getting cached value for ${name}...`);
    return getFileContents(path.join(cacheDirectory, name));
  }

  static set(name: string, value: object) {
    logger.verbose(`Saving new value to ${path.join(cacheDirectory, name)}...`); // eslint-disable-line
    saveToFile({
      directory: cacheDirectory,
      file: name,
      content: JSON.stringify(value),
    });
  }
}
