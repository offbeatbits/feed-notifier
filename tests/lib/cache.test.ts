import mock from 'mock-fs';
import { Cache } from '../../src/lib/cache';

const mockFs = (structure?: any) => mock(structure);

describe('Cache', () => {
  it('should implement get method', () => {
    expect(Cache.get).toBeDefined();
  });

  it('should implement set method', () => {
    expect(Cache.set).toBeDefined();
  });

  it('should cache value', () => {
    mockFs();

    Cache.set('foo1', {
      bar: 'baz',
    });
    expect(Cache.get('foo1')).toMatchSnapshot();
    mock.restore();
  });

  it('should not create cache directory if it exists', () => {
    const cacheDir = process.env.FN_CACHE_DIRECTORY!;
    mock({
      [cacheDir]: {},
    });

    Cache.set('foo2', {
      bar: 'baz',
    });

    expect(Cache.get('foo2')).toMatchSnapshot();
    mock.restore();
  });
});
