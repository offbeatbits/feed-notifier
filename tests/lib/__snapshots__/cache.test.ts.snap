// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cache should cache value 1`] = `"{\\"bar\\":\\"baz\\"}"`;

exports[`Cache should not create cache directory if it exists 1`] = `"{\\"bar\\":\\"baz\\"}"`;
